src/main/java/com/sk89q/craftbook/mechanics/pipe/PipeFilterEvent.java

```diff
@@ -0,0 +1,53 @@
+package com.sk89q.craftbook.mechanics.pipe;
+
+import org.bukkit.block.Block;
+import org.bukkit.event.HandlerList;
+import org.bukkit.inventory.ItemStack;
+
+import javax.annotation.Nonnull;
+import java.util.List;
+import java.util.Set;
+
+public class PipeFilterEvent extends PipeEvent {
+
+    private static final HandlerList handlers = new HandlerList();
+
+    private Set<ItemStack> filters;
+    private Set<ItemStack> exceptions;
+    private List<ItemStack> filteredItems;
+
+    public PipeFilterEvent(Block theBlock, List<ItemStack> items, Set<ItemStack> filters, Set<ItemStack> exceptions, List<ItemStack> filteredItems) {
+        super(theBlock, items);
+
+        this.filters = filters;
+        this.exceptions = exceptions;
+        this.filteredItems = filteredItems;
+    }
+
+    public Set<ItemStack> getFilters() {
```

Comment:
while filters/exceptions are the names used in code, from an API perspective they are kinda awful names. would you be able to please replace it with something like `getIncludeFilters` and `getExcludeFilters`?